### PR TITLE
Allowing multiple eventIds in TemporaryIdentity

### PR DESCRIPTION
**Pull request by Oluane**
## Description

This PR is a preparatory work for the implementation of multiple `eventId` support when verifying an identity. It changes the `eventId` key in DynamoDB's `temporaryIdentity` to be an array of strings. Changes have been made in queries and commands using it. 

## Type of change

- [x] **Chore** (non-breaking change which refactors / improves the existing code base).
- [ ] **Bug fix** (non-breaking change which fixes an issue).
- [ ] **New feature** (non-breaking change which adds functionality).
- [ ] **Breaking change** (fix or feature that would cause existing functionality to not work as expected).

## Checklist:

- [x] My code follows the style [**contributing guidelines**][contributing_file] of this project.
- [x] I have performed a self-review of my own code.
- [ ] I have commented my code in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.

[contributing_file]: https://github.com/fewlinesco/connect-account/blob/master/README.adoc

### CODE REVIEW

**Comment by Fenntasy** [comment]
Task linked: [CU-eaxnxa [feature] - Allow multiple event id for resend validation code](https://app.clickup.com/t/eaxnxa)

File: src/pages/api/auth-connect/verify-validation-code.ts [hunk 1]
```diff
@@ -69,21 +69,13 @@ const handler: Handler = async (request, response) => {
 
     span.setDisclosedAttribute("user found", true);
 
-    const temporaryIdentity = user.temporary_identities.find(
-      (temporaryIdentity) => temporaryIdentity.eventId === eventId,
-    );
+    const temporaryIdentity = user.temporary_identities.find(({ eventIds }) => {
+      return eventIds.find((eventId) => eventId === request.body.eventId);
+    });
```
Comment (Fuzznimp) [inline src/pages/api/auth-connect/verify-validation-code.ts:74]:
```suggestion
    const temporaryIdentity = user.temporary_identities.find(({ eventIds }) => {
      return eventIds.find((InDbEventId) => InDbEventId === eventId);
    });
```

What do you think?

File: src/pages/api/auth-connect/re-send-identity-validation-code.ts [hunk 2]
```diff
@@ -73,14 +73,16 @@ const handler: Handler = (request, response): Promise<void> => {
       }
 
       const temporaryIdentity = user.temporary_identities.find(
-        ({ eventId }) => eventId === request.body.eventId,
+        ({ eventIds }) => {
+          return eventIds.find((eventId) => eventId === request.body.eventId);
+        },
```
Comment (Fuzznimp) [inline src/pages/api/auth-connect/re-send-identity-validation-code.ts:78]: Same as `verify-validation-code`.

File: src/pages/api/auth-connect/verify-validation-code.ts (hunk 1 above)
Comment (Oluane) [reply src/pages/api/auth-connect/verify-validation-code.ts:74]: 👍

**Review by Fuzznimp — APPROVED** [review]

**Pull request merged**